src/linear_algebra/matrix/block.lean

```diff
@@ -114,14 +90,8 @@ begin
   { exact absurd hk12 (irrefl 1) }
 end
 
-/-- Let `b` map rows and columns of a square matrix `M` to blocks indexed by `ℕ`s. Then
-  `block_triangular_matrix M n b` says the matrix is block triangular. -/
-def block_triangular_matrix {o : Type*} (M : matrix o o R) (b : o → ℕ) : Prop :=
-∀ i j, b j < b i → M i j = 0
-
-lemma upper_two_block_triangular {m n : Type*}
-  (A : matrix m m R) (B : matrix m n R) (D : matrix n n R) :
-  block_triangular_matrix (from_blocks A B 0 D) (sum.elim (λ i, 0) (λ j, 1)) :=
+lemma upper_two_block_triangular (A : matrix m m R) (B : matrix m n R) (D : matrix n n R) :
+  block_triangular (from_blocks A B 0 D) (sum.elim (λ i, 0) (λ j, 1)) :=
```

Comment:
What type are the `0` and `1` here? This doesn't look to me like it's been generalized in the way descrived in the PR description.

Reply:
Good point. Done.